skaha/src/main/java/org/opencadc/skaha/session/SessionAction.java

```diff
@@ -237,7 +237,7 @@ public static String getContributedURL(String host, String sessionID) throws Mal
         return "https://" + host + "/session/contrib/" + sessionID + "/";
     }
     
-    protected void injectProxyCert(final Subject subject, String userid, String posixID)
```

Comment:
I think having the monitor on this method will cause proxy injection to be synchronized for all users, not just the one launching a new session.  Plus, since ~/.ssl is on arc they could be doing things in there without going through skaha.  I would suggest making the injection atomic by creating the file in /tmp/ and then moving it to ~/.ssl/cadcproxy.pem.

Reply:
Refactored SessionAction.injectProxyCert(...). Please take a look. Thanks.